bot/cogs/weather_cog.py

```diff
@@ -0,0 +1,206 @@
+            night_temp = res_json.get('daily',{})[i].get('temp',{}).get('night',{})
+            
+            # Chance of Precipitation
+            precp = res_json.get('daily',{})[i].get('pop',{})
+            
+            # Weather Conditions
+            desc = res_json.get('daily',{})[i].get('weather',{})[0].get('description',{}).title()
+
+            # Humidity
+            hum = res_json.get('daily',{})[i].get('humidity',{})
+
+            # Wind Speed and Direction
+            wind = res_json.get('daily',{})[i].get('wind_speed',{})
+            wind_deg = res_json.get('daily',{})[i].get('wind_deg',{})
+            ix = round(float(wind_deg) / (360. / len(dirs)))
+            wind_dir = dirs[ix % len(dirs)]
+
+            if i > 1:
+                page += f'Day {i}: {day_temp}°F\n'
+                page += f'Night {i}: {night_temp}°F\n'
+            elif i == 1:
+                page += f'Tomorrow: {day_temp}°F\n'
+                page += f'Tomorrow Night: {night_temp}°F\n'
+            else:
+                page += f'Today: {day_temp}°F\n'
+                page += f'Tonight: {night_temp}°F\n'
+            
+            page += f'Condition: {desc}\n\n'
+            page += f'Chance of Precipitation: {round(precp*100)}%\n'
+            page += f'Humidity: {hum}%\n'
+            page += f'Wind: {wind} mph ({wind_dir})\n\n'
+            pages.append(page)
+
+            '''
+            if i % 2 != 0 and i != 7:
+                pages.append(page)
+                page = ''
+            else:
+                page += '\n\n' 
+            '''
+        pages.append(page)
+        
+        return pages
+
+    @commands.command()
+    async def weather(self, ctx, loc):
+        """
+        Determine the weather at a given location
+
+        USE:
+        EXAMPLE:
+        """
+
+        # Remove any characters besides &, _, or - that are not in ranges a-z, A-Z, or 0-9
+        # Exceptions:
+        # & _ - and ,
+        # per the ASCII Table https://www.asciitable.com
+        loc = re.sub("[^a-zA-Z0-9 ,&_-]+", "", loc)
+
+        # Geocoding URL
+        url_GeoAPI = f'https://geocode.xyz/{loc}'
+
+        self.geocode_api_key = BotSecrets.get_instance().geocode_key
+        self.weather_api_key = BotSecrets.get_instance().weather_key
+
+        geo_queryparams = {
+            'auth' : self.geocode_api_key,
+            'json' : '1',
+            }
+
+        # Try Except for catching errors that could give away either API key
+        try:
+            async with aiohttp.request("GET", url_GeoAPI, params=geo_queryparams) as response:
+                if (response.status == 200):
+                    res_geo_json = await response.json()
+                    city = res_geo_json.get('standard',{}).get('city',{})
+                    lon = res_geo_json.get('longt',{})
+                    lat = res_geo_json.get('latt',{})
+                    
+                    url = "https://api.openweathermap.org/data/2.5/onecall"
```

Comment:
we should pull this out into a module level CONSTANT

Reply:
Is that just to make it easier to find and to be a constant for the entire cog?

Reply:
yee, better to have constants like api urls defined in conspicuous spots